app/assets/v2/js/pages/increase_bounty.js

```diff
@@ -105,53 +105,78 @@ $(document).ready(function() {
     }
 
     var bountyAmount = parseInt($('input[name=valueInToken]').val(), 10);
-    var ethAmount = isETH ? amount : 0;
-    var bountyId = $('input[name=standardBountiesId]').val();
-    var fromAddress = $('input[name=bountyOwnerAddress]').val();
 
-    var errormsg = undefined;
-
-    if (bountyAmount == 0 || open == false) {
-      errormsg = 'No active funded issue found at this address. Are you sure this is an active funded issue?';
-    }
-    if (fromAddress != web3.eth.coinbase) {
-      errormsg = 'Only the address that submitted this funded issue may increase the payout.';
-    }
-
-    if (errormsg) {
-      _alert({ message: errormsg });
-      unloading_button($('#submitBounty'));
-      return;
-    }
-
-    function approveSuccessCallback() {
-      bounty.increasePayout(
-        bountyId,
-        bountyAmount + amount,
-        amount,
-        {
-          from: account,
-          value: ethAmount,
-          gasPrice: web3.toHex($('#gasPrice').val()) * Math.pow(10, 9)
-        },
-        web3Callback
-      );
-    }
+    // Use DB to determine which networks each bounty is on.
+    var uri = '/api/v0.1/bounties/?github_url=' + issueURL;
+
+    $.get(uri, function(results, status) {
+      results = sanitizeAPIResults(results);
+      result = results[0];
+      if (result == null) {
+        _alert({
+          message: 'No active bounty found for this Github URL.'
+        });
+        unloading_button($('.js-submit'));
+        return;
+      }
 
-    if (isETH) {
-      // no approvals needed for ETH
-      approveSuccessCallback();
-    } else {
-      token_contract.approve(
-        bounty_address(),
-        amount,
-        {
-          from: account,
-          value: 0,
-          gasPrice: web3.toHex($('#gasPrice').val()) * Math.pow(10, 9)
-        },
-        approveSuccessCallback
-      );
-    }
+      var bountyNetwork = result['network'];
+
+      // Determine if browser and bounty networks match
+      web3.version.getNetwork((error, netId) => {
```

Comment:
i wonder if theres a way to DRY for this check across multiple of these JS files

Reply:
Sorry, not sure I follow.. Are you asking if it can be better factored out into a shared function?

Reply:
yes. DRY = dont repeat yourself